src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -426,29 +426,29 @@ void primitive_inst::set_arguments() {
 }
 
 void primitive_inst::build_deps() {
-    if (_deps.empty() && !_node.get_dependencies().empty()) {
-        _deps = _network.get_primitives(_node.get_dependencies());
+    if (_deps.empty() && !_node->get_dependencies().empty()) {
+        _deps = _network.get_primitives(_node->get_dependencies());
         _exec_deps = build_exec_deps(_deps);
     }
-    if (_deps_new.empty() && !_node.get_dependencies_new().empty()) {
-        _deps_new = _network.get_primitives(_node.get_dependencies_new());
+    if (_deps_new.empty() && !_node->get_dependencies_new().empty()) {
+        _deps_new = _network.get_primitives(_node->get_dependencies_new());
     }
 }
 
-primitive_inst::primitive_inst(network& network, program_node const& node, bool allocate_memory)
+primitive_inst::primitive_inst(network& network, std::shared_ptr<const program_node> node, bool allocate_memory)
```

Comment:
nit but program::ptr ?

Reply:
This is not `program` but `program_node`.